flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreWriteImpl.java

```diff
@@ -34,30 +36,38 @@
 import org.apache.flink.table.store.file.mergetree.compact.CompactStrategy;
 import org.apache.flink.table.store.file.mergetree.compact.MergeFunction;
 import org.apache.flink.table.store.file.mergetree.compact.UniversalCompaction;
+import org.apache.flink.table.store.file.stats.FileStatsExtractor;
 import org.apache.flink.table.store.file.utils.FileStorePathFactory;
 import org.apache.flink.table.store.file.utils.RecordReaderIterator;
+import org.apache.flink.table.store.file.writer.AppendOnlyWriter;
 import org.apache.flink.table.store.file.writer.RecordWriter;
 import org.apache.flink.table.types.logical.RowType;
 
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.concurrent.ExecutorService;
 import java.util.function.Supplier;
-import java.util.stream.Collectors;
 
 /** Default implementation of {@link FileStoreWrite}. */
 public class FileStoreWriteImpl implements FileStoreWrite {
 
     private final DataFileReader.Factory dataFileReaderFactory;
     private final DataFileWriter.Factory dataFileWriterFactory;
+    private final WriteMode writeMode;
+    private final FileFormat fileFormat;
+    private final RowType writeSchema;
     private final Supplier<Comparator<RowData>> keyComparatorSupplier;
     private final MergeFunction mergeFunction;
     private final FileStorePathFactory pathFactory;
     private final FileStoreScan scan;
     private final MergeTreeOptions options;
 
     public FileStoreWriteImpl(
+            WriteMode writeMode,
+            RowType writeSchema,
             RowType keyType,
             RowType valueType,
```

Comment:
What is this `writeSchema` for? We already have a `valueType`.

Reply:
The `writeSchema` can be removed now.  In the previous version, I was thinking that valueType would be some customized table schema (instead of the original table schema), such as `int _VALUE_COUNT` or some other type. But in fact, for append-only table the `valueType` will always be `tableSchema`.  So we don't have an extra `writeSchema` any more. Thanks for the comment.